src/services/swap_service.ts

```diff
@@ -69,13 +71,18 @@ export class SwapService {
             from,
             excludedSources,
             affiliateAddress,
+            apiKey,
+            intentOnFilling,
         } = params;
         const assetSwapperOpts = {
             ...ASSET_SWAPPER_MARKET_ORDERS_OPTS,
             slippagePercentage,
             bridgeSlippage: slippagePercentage,
             gasPrice: providedGasPrice,
             excludedSources, // TODO(dave4506): overrides the excluded sources selected by chainId
+            apiKey,
+            intentOnFilling,
+            takerAddress: from,
```

Comment:
Consider not conflating `from` and `takerAddress`. As you discovered, this is sometimes not often the case. I.e Via forwarder `takerAddress` is the Forwarder, from is the end user.

What happens when the Forwarder is used? the `takerAddress` in the order MUST be the Forwarder contract.

Reply:
I tried to account for this here: https://github.com/0xProject/0x-api/pull/162/commits/0255b25dcc8a9cefccfccdbcb50898dcd9ee9ef2. Does that address the issue?